### PR TITLE
Exit status and signal improvement

### PR DESCRIPTION
**Pull request by kormat**
- Use SIGINT to signal there's an error.
- Check errors properly in integration_test.sh
- Setup signal handling in end2end
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/314%23issuecomment-130316572%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36856803%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36856901%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36856993%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36863203%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36863270%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36867900%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36867901%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36867902%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23issuecomment-130326885%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23issuecomment-130354859%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23issuecomment-130316572%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Tests%20failed%20due%20to%20%23316.%22%2C%20%22created_at%22%3A%20%222015-08-12T14%3A07%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22lgtm%22%2C%20%22created_at%22%3A%20%222015-08-12T14%3A44%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Tests%20fixed%20now%2C%20merging.%22%2C%20%22created_at%22%3A%20%222015-08-12T15%3A58%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2007cd74773a5ac9e3dbb1a71595823342f5d77c92%20test/lib_util_test.py%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36856993%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22SIGINT%20again%3F%20and%20for%20the%20import%3F%22%2C%20%22created_at%22%3A%20%222015-08-12T13%3A10%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-12T14%3A44%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/lib_util_test.py%3AL311-328%22%7D%2C%20%22Pull%2007cd74773a5ac9e3dbb1a71595823342f5d77c92%20lib/util.py%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36856901%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22SIGINT%3F%22%2C%20%22created_at%22%3A%20%222015-08-12T13%3A09%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22%60SIGTERM%60%20is%20the%20normal%20shutdown%20signal%2C%20sent%20from%20supervisor.%20Anything%20else%20is%20an%20error%2C%20and%20is%20handled%20differently.%20This%20is%20why%20%60kill_self%60%20now%20sends%20%60SIGINT%60%20instead.%22%2C%20%22created_at%22%3A%20%222015-08-12T14%3A06%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-12T14%3A44%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/util.py%3AL303-316%22%7D%2C%20%22Pull%2007cd74773a5ac9e3dbb1a71595823342f5d77c92%20lib/thread.py%202%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/314%23discussion_r36856803%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22SIGINT%22%2C%20%22created_at%22%3A%20%222015-08-12T13%3A08%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Fixed%22%2C%20%22created_at%22%3A%20%222015-08-12T14%3A07%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-12T14%3A44%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/thread.py%3AL30-37%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/314#issuecomment-130316572'>General Comment</a></b>
- <a href='https://github.com/kormat'></a> Tests failed due to #316.
- <a href='https://github.com/pszalach'></a> lgtm
- <a href='https://github.com/kormat'></a> Tests fixed now, merging.
- [x] <a href='#crh-comment-Pull 07cd74773a5ac9e3dbb1a71595823342f5d77c92 lib/thread.py 2'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/314#discussion_r36856803'>File: lib/thread.py:L30-37</a></b>
- <a href='https://github.com/pszalach'></a> SIGINT
- <a href='https://github.com/kormat'></a> Fixed
- [x] <a href='#crh-comment-Pull 07cd74773a5ac9e3dbb1a71595823342f5d77c92 lib/util.py 7'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/314#discussion_r36856901'>File: lib/util.py:L303-316</a></b>
- <a href='https://github.com/pszalach'></a> SIGINT?
- <a href='https://github.com/kormat'></a> `SIGTERM` is the normal shutdown signal, sent from supervisor. Anything else is an error, and is handled differently. This is why `kill_self` now sends `SIGINT` instead.
- [x] <a href='#crh-comment-Pull 07cd74773a5ac9e3dbb1a71595823342f5d77c92 test/lib_util_test.py 14'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/314#discussion_r36856993'>File: test/lib_util_test.py:L311-328</a></b>
- <a href='https://github.com/pszalach'></a> SIGINT again? and for the import?

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/314?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/314?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/314'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: lib/thread.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def kill_self():
     """
     Send SIGTERM to self, to allow quitting the process from threads.
```
Comment (pszal) [inline lib/thread.py]: SIGINT

File: lib/util.py [hunk 2]
```diff
@@ -303,8 +303,13 @@ def _signal_handler(signum, _):
     :param signum:
     :type signum:
     """
-    logging.info("Received %s", _SIG_MAP[signum])
-    sys.exit(0)
+    text = "Received %s" % _SIG_MAP[signum]
+    if signum == signal.SIGTERM:
```
Comment (pszal) [inline lib/util.py:314]: SIGINT?

File: test/lib_util_test.py [hunk 3]
```diff
@@ -310,10 +311,17 @@ class TestSignalHandler(object):
     @patch("lib.util.sys.exit", autospec=True)
     @patch("lib.util.logging.info", autospec=True)
     def test_basic(self, info, exit):
-        _signal_handler(1, 2)
-        info.assert_called_once_with("Received %s", _SIG_MAP[1])
+        _signal_handler(SIGTERM, "")
```
Comment (pszal) [inline test/lib_util_test.py:328]: SIGINT again? and for the import?

File: lib/util.py (hunk 2 above)
Comment (kormat) [inline lib/util.py:314]: `SIGTERM` is the normal shutdown signal, sent from supervisor. Anything else is an error, and is handled differently. This is why `kill_self` now sends `SIGINT` instead.

File: lib/thread.py (hunk 1 above)
Comment (kormat) [inline lib/thread.py]: Fixed

File: test/lib_util_test.py (hunk 3 above)
Comment (pszal) [inline test/lib_util_test.py:328]: :+1:<a href='#crh-update-none'></a>

File: lib/util.py (hunk 2 above)
Comment (pszal) [inline lib/util.py:314]: :+1:<a href='#crh-update-none'></a>

File: lib/thread.py (hunk 1 above)
Comment (pszal) [inline lib/thread.py]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**